address.go

```diff
@@ -0,0 +1,18 @@
+package ie
+
+import "github.com/intervention-engine/fhir/models"
+
+type RestructedAddress struct {
```

Comment:
With the restructuring of these types, we have a real opportunity here to hide some of the nastiness of fhir from the ie app. From the looks of this PR, the only code that really _needs_ to know what the fhir structure looks like is the mongo code. So we could move the fhir representations into the mongo package along with their constructors and have the mongo code deal with the transformations. This would also allow the ie package to keep concise names for the types like "Address" and "Patient" without any confusion and we get better type names out of it.